test/gtest/ucp/test_ucp_fence.cc

```diff
@@ -113,10 +124,15 @@ class test_ucp_fence : public ucp_test {
                      ucp_rkey_h rkey, void *memheap_ptr,
                      uint64_t initial_value, uint32_t* error) {
         ucs::ptr_vector<worker> m_workers;
+        std::vector<entity*> entities;
         m_workers.clear();
         m_workers.push_back(new worker(this, send1, send2, sender, rkey,
                                        memheap_ptr, initial_value, error));
-        m_workers.at(0).join();
+        entities.push_back(&receiver());
+        while (m_workers.at(0).join_np()) {
```

Comment:
1. can we count running threads in the test class (atomic var for example) instead of using pthread_tryjoin_np?
2. maybe use pthread barrier to sync threads before they exit?

Reply:
re-designed test